### ISSUE TITLE
🎁: Add compile errors to quickfix list on failed build

### ISSUE THREAD

**Issue by SchnozzleCat** [GustavEikaas/easy-dotnet.nvim#21]
Something that I am still looking for in an nvim plugin is the ability to quickly fix build issues on e.g. a large refactor. 

Something like [this](https://github.com/dmmulroy/tsc.nvim) exists for typescript, which pushes all issues to the quickfix list. If a dotnet build fails, I think it would be neat to have those issues in the quickfix list for faster iteration.

**Comment by GustavEikaas** [comment]
Interesting idea, I tried to make a simple POC for this, my biggest obstacle is getting structured output from the dotnet build command. The terminal output would be hard to parse without making a complex web of regex parsers. 
The following command gave me some pretty solid output which seems like it could be parsed and entered into a quickfix list rather easily.
` dotnet build /flp:v=q /flp:logfile=MyLog.log`

**Comment by SchnozzleCat** [comment]
I'd also be happy with enabling workspace diagnostics, but I use the new-ish ``roslyn``-based LSP instead of omnisharp, and while it does support workspace diagnostics, I cannot get them to work with ``roslyn.nvim``. 

Being able to quickly fix issues on build would be a viable alternative IMO, so thanks for looking into it!

**Comment by GustavEikaas** [comment]
Made a minimal version of this, seems to be working for me atleast, not tested extensively. I need more time to figure out how I want to incorporate this into this project. Do you want to test my minimal version? Just paste this code into the configuration

```lua
local logPath = vim.fn.stdpath "data" .. "/easy-dotnet/build.log"
    local dotnet = require("easy-dotnet")
    dotnet.setup({
      terminal = function(path, action)
        local commands = {
          run = function()
            return "dotnet run --project " .. path
          end,
          test = function()
            return "dotnet test " .. path
          end,
          restore = function()
            return "dotnet restore " .. path
          end,
          build = function()
            return "dotnet build " .. path .. " /flp:v=q /flp:logfile=" .. logPath
          end
        }
        local command = commands[action]() .. "\r"
        require("toggleterm").exec(command, nil, nil, nil, "float")
      end,
    })

    vim.api.nvim_create_user_command("T", function()
      populate_quickfix_from_file(logPath)
    end, {})

```

**Comment by SchnozzleCat** [comment]
> ```lua
> local logPath = vim.fn.stdpath "data" .. "/easy-dotnet/build.log"
> ```

Where is the ``populate_quickfix_from_file`` function coming from?

The output of the build command is being written to the file correctly though, so this approach should work as long as it can be filtered reliably. Very cool!

**Comment by GustavEikaas** [comment]
Woops my bad.

```lua
local function populate_quickfix_from_file(filename)
  -- Open the file for reading
  local file = io.open(filename, "r")
  if not file then
    print("Could not open file " .. filename)
    return
  end

  -- Table to hold quickfix list entries
  local quickfix_list = {}

  -- Iterate over each line in the file
  for line in file:lines() do
    -- Match the pattern in the line
    local filepath, lnum, col, text = line:match("^(.+)%((%d+),(%d+)%)%: (.+)$")

    if filepath and lnum and col and text then
      -- Remove project file details from the text
      text = text:match("^(.-)%s%[.+$")

      -- Add the parsed data to the quickfix list
      table.insert(quickfix_list, {
        filename = filepath,
        lnum = tonumber(lnum),
        col = tonumber(col),
        text = text,
      })
    end
  end

  -- Close the file
  file:close()

  -- Set the quickfix list
  vim.fn.setqflist(quickfix_list)

  -- Open the quickfix window
  vim.cmd("copen")
end



```

**Comment by SchnozzleCat** [comment]
That works! Is there a way to run it if the build command does not return code 0?

**Comment by GustavEikaas** [comment]
@SchnozzleCat This has now been implemented, feel free to test it out. Its exposed as build_quickfix, example below
```lua
    vim.api.nvim_create_user_command('Build', function()
      dotnet.build_quickfix()
    end, {})
```
